tools/decode.py

```diff
@@ -37,100 +45,62 @@ def run(file, **kwargs):
-                        # Read our servo
-                        servo = msg.sensors.servo.rAnkleRoll
-
-                        if prev_goal_position == None:
-                            prev_goal_position = round(servo.goalPosition)
-                        elif prev_goal_position != round(servo.goalPosition):
-                            # Changing motion
-                            if state != None:
-                                pickle.dump((state, c_pass), of, protocol=pickle.HIGHEST_PROTOCOL)
-                                c_pass = []
-
-                            if prev_goal_position == 0:
-                                # Second half
-                                if round(servo.goalPosition) > prev_goal_position:
-                                    # increasing
-                                    state = 'inc2stop'
-                                else:
-                                    # decreasing
-                                    state = 'dec2stop'
-                            elif round(servo.goalPosition) == 0:
-                                # First half
-                                if round(servo.goalPosition) > prev_goal_position:
-                                    # increasing
-                                    state = 'inc1stop'
-                                else:
-                                    # decreasing
-                                    state = 'dec1stop'
-                            elif round(servo.goalPosition) > prev_goal_position:
-                                # increasing
-                                state = 'inc'
-                            else:
-                                # decreasing
-                                state = 'dec'
-
-                            prev_goal_position = round(servo.goalPosition)
-
-                        c_pass.append({
-                            'error_flags': servo.errorFlags,
-                            'present_position': servo.presentPosition,
-                            'goal_position': servo.goalPosition,
-                            'present_speed': servo.presentSpeed,
-                            'goal_speed': servo.movingSpeed,
-                            'load': servo.load,
-                            'voltage': servo.voltage,
-                            'temperature': servo.temperature
-                        })
-
-                    else:
-                        out = re.sub(r'\s+', ' ', MessageToJson(msg, True))
-                        out = '{{ "type": "{}", "timestamp": {}, "data": {} }}'.format(parsers[type_hash][0].decode('utf-8'), timestamp, out)
-                        # Print as a json object
-                        print(out)
 
+        # NBS File Format:
+        # 3 Bytes - NUClear radiation symbol header, useful for synchronisation when attaching to a existing stream.
+        # 4 Bytes - The remaining packet length i.e. 16 bytes + N payload bytes
```

Comment:
And by missing period you mean too many periods! (no other comments in the file have them)